### PR TITLE
Add summary table for Sphinx RST files

### PR DESCRIPTION
**Pull request by fwmeng88**
This PR mainly:
- [x] Add a summary table to Sphinx documentation,  987920d
- [x] Update references, 51f0bd5
- [x] Add internal links to APIs, 4212325
- [x] Ignore the RST style checking for `intro_about.rst` file as I didn't figure out a better way of building the table, which ends up with long lines. But this should not matter a lot since this file is just documentation for the website and everything is clear for the website.

All the tests passed.

### CODE REVIEW

File: docs/intro_about.rst [hunk 1]
```diff
@@ -56,6 +56,72 @@ matrices, rotation matrices, symmetric matrices, or permutation matrices. When :
...
++--------------------------------------------------------------------------------------------------------------------+------------------------------+----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| :ref:`Two-sided Orthogonal <orthogonal>` :cite:`two_sided_ortho:1`                                                 | :math:`\mathbf{Q}_1`         | :math:`\mathbf{Q}_2` | :math:`\begin{cases} \mathbf{Q}_1^{-1} = \mathbf{Q}_1^\dagger \\ \mathbf{Q}_2^{-1} = \mathbf{Q}_2^\dagger \\ \end{cases}`                                                                                                                                       |
```
Comment (FarnazH) [inline docs/intro_about.rst:73]: Change `:math:`\mathbf{Q}_1` to `:math:`\mathbf{Q}_1^\dagger` (I changed the table in the paper a little while ago, it makes things simpler to keep the dagger for U1)

File: docs/intro_about.rst [hunk 2]
```diff
@@ -56,6 +56,72 @@ matrices, rotation matrices, symmetric matrices, or permutation matrices. When :
...
++--------------------------------------------------------------------------------------------------------------------+------------------------------+----------------------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
+| :ref:`Two-sided Permutation <permutation>` :cite:`two_sided_permutation:1`                                         | :math:`\mathbf{P}_1`         | :math:`\mathbf{P}_2` | :math:`\begin{cases} [\mathbf{P}_1]_{ij} \in \{0, 1\} \\ [\mathbf{P}_2]_{ij} \in \{0, 1\} \\ \sum_{i=1}^n [\mathbf{P}_1]_{ij} = \sum_{j=1}^n [\mathbf{P}_1]_{ij} = 1 \\ \sum_{i=1}^n [\mathbf{P}_2]_{ij} = \sum_{j=1}^n [\mathbf{P}_2]_{ij} = 1 \\ \end{cases}` |
```
Comment (FarnazH) [inline docs/intro_about.rst:77]: Change `:math:`\mathbf{P}_1` to `:math:`\mathbf{P}_1^\dagger` (I changed the table in the paper a little while ago, it makes things simpler to keep the dagger for P1)

**Review by FarnazH — CHANGES_REQUESTED** [review]

File: docs/intro_about.rst (hunk 2 above)
Comment (fwmeng88) [reply docs/intro_about.rst]: The issue is fixed in 860b02b.

File: docs/intro_about.rst (hunk 1 above)
Comment (fwmeng88) [reply docs/intro_about.rst]: It's fixed in 860b02b.

**Review by FarnazH — APPROVED** [review]

**Pull request merged**